src/utils/api.js

```diff
@@ -39,4 +39,18 @@ return response.data.articles;
          return response.data.comments;
                 });
         };
-    
\ No newline at end of file
+    
+        export const fetchTopics = () => {
+            return  newsApi.get(`/topics`).then(({data})=>{
+               
+                return data.topics
+            })
+        }
+
+        export const fetchTopicArticles = (topic) => {
+            return newsApi.get(`/articles/?topic=${topic}`).then((response) =>{
+          
+             
+                return response.data.articles;
+            })
```

Comment:
Do you need a new function or could you of refactord your previous fetchArticles to take a topic param.